R/length-params.R

```diff
@@ -1,68 +1,165 @@
+#'
+#' @return No output. This funcrion is used for it's side effects
+#' @export
+#'
+get_catch_types <- function(log_threshold = logger::DEBUG) {
   pars <- read_config()
 
-  fao_records <- air_get_records(table="fao_catch",
-                                 base_id=pars$metadata$airtable$base_id,
-                                 api_key=readLines("auth/airtable-key")) %>%
-    air_records_to_tibble()
+  metadata <- get_preprocessed_metadata(pars)
 
-  # get sp ranked table by isscaap_taxocode
-  timor_sp <- dplyr::filter(fao_records, !grepl("X",isscaap_taxocode) &
-                              !is.na(isscaap_code))
+  catch_types <- metadata$catch_types
+  fao_records <- metadata$fao_catch
 
-  # retrieve length params
-  params_sp <-
-    rfishbase::length_length(timor_sp$name_scientific) %>%
-    dplyr::group_by(Species) %>%
-    dplyr::arrange(desc(Number)) %>% # arrange by number of records and take max
-    dplyr::slice(1)
-
-  params_sp
-}
+  catches_tab <- dplyr::left_join(fao_records, catch_types, by = "interagency_code")
 
+  spp <- catches_tab %>%
+    dplyr::select(
+      .data$interagency_code, .data$isscaap_taxocode.x, .data$name_scientific,
+      .data$catch_family, .data$catch_name_en, .data$taxo_name_en
+    )
 
-length_params_spp <- function(log_threshold = logger::DEBUG){
-
-  logger::log_threshold(log_threshold)
-
-  pars <- read_config()
+  logger::log_info("Getting taxa ranks...")
+  ranks <- taxize::tax_rank(spp$name_scientific, db = "gbif", rows = 1)
 
-  fao_records <- air_get_records(table="fao_catch",
-                                 base_id=pars$metadata$airtable$base_id,
-                                 api_key=readLines("auth/airtable-key")) %>%
-    air_records_to_tibble()
-
-  # get spp ranked table by isscaap_taxocode
-  spp <- dplyr::filter(fao_records, grepl("X",isscaap_taxocode) &
-                               !is.na(isscaap_code))
-
-  spp$name_scientific <- stringr::word(spp$name_scientific, 1)
-
-  # Use taxize::tax_rank to get taxonomical rank
-  ranks <- taxize::tax_rank(spp$name_scientific,db = "gbif",rows=1)
 
   ranks_tab <-
     ranks %>%
-    purrr::map(~stringr::str_to_title(.x)) %>%
-    purrr::map(~tidyr::as_tibble(.x)) %>%
-    purrr::map_dfr(~dplyr::mutate_all(.x,as.character)) %>%
-    dplyr::rename(taxa_rank=value) %>%
+    purrr::map(~ stringr::str_to_title(.x)) %>%
+    purrr::map(~ tidyr::as_tibble(.x)) %>%
+    purrr::map_dfr(~ dplyr::mutate_all(.x, as.character)) %>%
+    dplyr::rename(taxa_rank = .data$value) %>%
     dplyr::bind_cols(spp) %>%
-    dplyr::select(taxa_rank,name_scientific)
+    dplyr::mutate(name = gsub(" spp", "", .data$name_scientific)) %>%
+    dplyr::mutate(name = gsub("\\s*\\([^\\)]+\\)", "", .data$name)) %>%
+    dplyr::mutate(
+      name = dplyr::case_when(
+        name == "Thunnini" ~ "Tuna",
+        name == "Selachimorpha" ~ "Shark",
+        name == "Hyporhamphus quoyi" ~ "Garfish",
+        name == "Abudefduf vaigiensis" ~ "Sergeant",
```

Comment:
Wondering why we need to do this? Is it because Fishbase is struggling to find a/b parameters for these species or because we we want to get parameters for other Garfishes and Sergeant fishes?